compiler/z/codegen/BinaryEvaluator.cpp

```diff
@@ -2466,8 +2466,36 @@ genericRotateAndInsertHelper(TR::Node * node, TR::CodeGenerator * cg)
    return NULL;
    }
 
+/*
+ *
+ * This is a helper function that will try to use a rotate instruction
+ * to perform a 'land' operation. For example, if we have the following tree:
+ *
+ * land
+ *    i2l
+ *       iload parm=0
+ *    lconst 0xFFFFFFFFF0000000
+ *
+ * We can avoid the 'land' operation altogether in such a case. We know that
+ * the result of this operation will preserve the first 16 bits of the iload
```

Comment:
> will preserve the first 16 bits of the iload

This does not sound correct. This example is kind of an edge case. I would focus more on the intention of the use of `RISBG` to avoid two AND instructions, i.e. the case where you are ANDing with something like `0xFF00000000000003` where there is a contiguous sequence of one bits somewhere in the constant and where the alternative to evaluating such a sequence is to either materialize the constant in a register (expensive) or to use two ANDs such as `NIHF` + `NILF` to and the most-significant and least-significant 32-bit portions of the first operand.